src/System.Collections.NonGeneric/tests/Hashtable/RemoveTests.cs

```diff
@@ -0,0 +1,288 @@
+            };
+
+            string[] strSecretIdentities = new string[]
+            {
+                "Batson, Billy" ,       //0
+                "Gordan, Barbara" ,     //1
+                "Grayson, Dick" ,       //2
+                "Jordan, Hal" ,     //3
+                "Drake, Tim" ,          //4
+                "Kent, Clark" ,     //5
+                "Lance, Dinah" ,        //6
+                "Parker, Peter" ,       //7
+                "Stark, Tony" ,     //8
+                "Troy, Donna" ,     //9
+                "Wayne, Bruce" ,        //10
+                "West, Wally" ,     //11
+                "Queen, Oliver" ,       //12
+                "Palmer, Ray" ,     //13
+                "Irons, John Henry" ,   //14
+                "Cage, Luke" ,          //15
+            };
+            #endregion
+
+            // Allocate the hash table.
+            hash = new Hashtable();
+
+            // Construct the hash table by adding items to the table.
+            for (ii = 0; ii < strSuperHeroes.Length; ++ii)
+            {
+                hash.Add(strSuperHeroes[ii], strSecretIdentities[ii]);
+            }
+
+            // Validate additions to Hashtable.
+            Assert.Equal(strSuperHeroes.Length, hash.Count);
+
+            //
+            // []Remove: Attempt to remove a bogus key entry from table.
+            //
+            hash.Remove("THIS IS A BOGUS KEY");
+            Assert.Equal(hash.Count, strSuperHeroes.Length);
+
+            //
+            // []	Remove: Attempt to remove a null key entry from table.
+            //
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                hash.Remove(null);
+            }
+            );
+
+            //
+            // [] Remove: Add key/value pair to Hashtable and remove items.
+            //
+            // Remove items from Hashtable.
+            for (ii = 0; ii < strSuperHeroes.Length; ++ii)
+            {
+                hash.Remove(strSuperHeroes[ii]);
+                Assert.Equal(strSuperHeroes.Length - ii - 1, hash.Count);
+            }
+
+
+            // now ADD ALL THE Entries the second time and remove them again
+            for (ii = 0; ii < strSuperHeroes.Length; ++ii)
+            {
+                hash.Add(strSuperHeroes[ii], strSecretIdentities[ii]);
+            }
+
+            // remove elements
+            for (ii = 0; ii < strSuperHeroes.Length; ++ii)
+            {
+                hash.Remove(strSuperHeroes[ii]);
+                Assert.Equal(hash.Count, strSuperHeroes.Length - ii - 1);
+            }
+
+            //[]Repeated removed
+            hash.Clear();
+            for (int iAnnoying = 0; iAnnoying < 10; iAnnoying++)
+            {
+                for (ii = 0; ii < strSuperHeroes.Length; ++ii)
+                {
+                    hash.Remove(strSuperHeroes[ii]);
+                    Assert.Equal(0, hash.Count);
+                }
+            }
+        }
+    }
+
+    class HashConfuse
```

Comment:
Also, there is more than one of the hash-collision-inducing helper class.  Could it be shared instead of multiply implemented?